Sources/FoundationEssentials/Calendar/Calendar_Gregorian.swift

```diff
@@ -258,30 +262,259 @@ internal final class _CalendarGregorian: _CalendarProtocol, @unchecked Sendable
+            }
+        case .nanosecond:
+            return maximumRange(of: smaller)
+        default:
+            break // Continue search
+        }
+
+        switch larger {
+        case .era:
+            // assume it cycles through every possible combination in an era at least once; this is a little dodgy for the Japanese calendar but this calculation isn't terribly useful either
+            switch smaller {
+            case .year, .quarter, .month, .weekOfYear, .weekOfMonth, .day:
+                return maximumRange(of: smaller)
+            case .weekdayOrdinal:
+                guard let r = maximumRange(of: .day) else { return nil }
+                return 1..<(((r.lowerBound + (r.upperBound - r.lowerBound) - 1 + 6) / 7) + 1)
+            default:
+                break
+            }
+        case .year:
+            switch smaller {
+            case .month:
+                return 1..<13
+            case .quarter, .weekOfYear: /* deprecated week */
+                return _algorithmA(smaller: smaller, larger: larger, at: capped)
+            case .day:
+                let year = dateComponent(.year, from: date)
+                let max = gregorianYearIsLeap(year) ? 366 : 365
+                return 1 ..< max + 1
+            case .weekOfMonth, .weekdayOrdinal:
+                return _algorithmB(smaller: smaller, larger: larger, at: capped)
+            default:
+                break
+            }
+        case .yearForWeekOfYear:
+            switch smaller {
+            case .quarter, .month, .weekOfYear:
+                return _algorithmA(smaller: smaller, larger: larger, at: capped)
+            case .weekOfMonth:
+                break
+            case .day, .weekdayOrdinal:
+                return _algorithmB(smaller: smaller, larger: larger, at: capped)
+            default:
+                break
+            }
+        case .quarter:
+            switch smaller {
+            case .month, .weekOfYear: /* deprecated week */
+                return _algorithmC(smaller: smaller, larger: larger, at: capped)
+            case .weekOfMonth, .day, .weekdayOrdinal:
+                return _algorithmB(smaller: smaller, larger: larger, at: capped)
+            default:
+                break
+            }
+        case .month:
+            switch smaller {
+            case .weekOfYear: /* deprecated week */
+                return _algorithmC(smaller: smaller, larger: larger, at: capped)
+            case .weekOfMonth, .day, .weekdayOrdinal:
+                return _algorithmA(smaller: smaller, larger: larger, at: capped)
+            default:
+                break
+            }
+        case .weekOfYear:
+            break
+        case .weekOfMonth: /* deprecated week */
+            switch smaller {
+            case .day:
+                return _algorithmD(at: capped)
+            default:
+                break
+            }
+        default:
+            break
+        }
+
+        return nil
     }
```

Comment:
I do want to factor this out, but I'm totally ok with landing this as-is until we figure out how to do that.